### ISSUE TITLE
runtime error in debug mode

### ISSUE THREAD

**Issue by cslxiao** [alexgkendall/caffe-segnet#23]
When  I run the training demo (CamVid) in the release mode, it works. But when I run in the debug model, it generates the following checking failure error:
__
I0225 10:53:33.101868  3843 net.cpp:235] This network produces output accuracy
I0225 10:53:33.101879  3843 net.cpp:235] This network produces output loss
I0225 10:53:33.101889  3843 net.cpp:235] This network produces output per_class_accuracy
I0225 10:53:33.101984  3843 net.cpp:482] Collecting Learning Rate and Weight Decay.
I0225 10:53:33.102026  3843 net.cpp:247] Network initialization done.
I0225 10:53:33.102035  3843 net.cpp:248] Memory required for data: 4332441652
I0225 10:53:33.102377  3843 solver.cpp:42] Solver scaffolding done.
I0225 10:53:33.102727  3843 solver.cpp:250] Solving VGG_ILSVRC_16_layer
I0225 10:53:33.102740  3843 solver.cpp:251] Learning Rate Policy: step
I0225 10:53:33.151669  3849 dense_image_data_layer.cpp:181] Prefetch batch: 41 ms.
I0225 10:53:33.151747  3849 dense_image_data_layer.cpp:182]      Read time: 26.489 ms.
I0225 10:53:33.151759  3849 dense_image_data_layer.cpp:183] Transform time: 14.965 ms.
**F0225 10:53:33.680835  3843 accuracy_layer.cpp:72] Check failed: label_value < num_labels (11 vs. 11) 
****\* Check failure stack trace: ***
    @     0x7f98fe11edaa  (unknown)
    @     0x7f98fe11ece4  (unknown)
    @     0x7f98fe11e6e6  (unknown)
    @     0x7f98fe121687  (unknown)
    @     0x7f98fe56a015  caffe::AccuracyLayer<>::Forward_cpu()
    @     0x7f98fe569183  caffe::Layer<>::Forward_gpu()
    @           0x423024  caffe::Layer<>::Forward()
    @     0x7f98fe605cad  caffe::Net<>::ForwardFromTo()
    @     0x7f98fe605a75  caffe::Net<>::ForwardPrefilled()
    @     0x7f98fe605e40  caffe::Net<>::Forward()
    @     0x7f98fe606755  caffe::Net<>::ForwardBackward()
    @     0x7f98fe6212bb  caffe::Solver<>::Step()
    @     0x7f98fe620cc5  caffe::Solver<>::Solve()
    @           0x41e9a8  train()
    @           0x42095c  main
    @     0x7f98fd934ec5  (unknown)
    @           0x41dda9  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)_
_

Anyone with the same problem and how can I resolve it?

**Comment by cslxiao** [comment]
I changed the assertions  in accuracy_layer.cpp:72 as follows:
**DCHECK_LT-->DCHECK_LE**
and it runs  in the debug mode. 

I  see that the loss_para is setted with ignore_label = 11 in the prototxt.  Then why this error happens during  debugging? Is that an bug? 
I am new to caffe. Can anyone give a explanation?

**Comment by minemebarsha** [comment]
Same here,

**F0512 01:45:15.311633 15462 accuracy_layer.cpp:72] Check failed: label_value < num_labels (11 vs. 11)** 
**\* Check failure stack trace: ***
    @     0x2b7be5acbb7d  google::LogMessage::Fail()
    @     0x2b7be5acdc7f  google::LogMessage::SendToLog()
    @     0x2b7be5acb76c  google::LogMessage::Flush()
    @     0x2b7be5ace51d  google::LogMessageFatal::~LogMessageFatal()
I0512 01:45:15.339488 15657 dense_image_data_layer.cpp:181] Prefetch batch: 482 ms.
I0512 01:45:15.339550 15657 dense_image_data_layer.cpp:182]      Read time: 31.611 ms.
I0512 01:45:15.339567 15657 dense_image_data_layer.cpp:183] Transform time: 12.371 ms.
    @     0x2b7be4b1e7c1  caffe::AccuracyLayer<>::Forward_cpu()
    @     0x2b7be4a979b1  caffe::Layer<>::Forward_gpu()
    @           0x415b74  caffe::Layer<>::Forward()
    @     0x2b7be4b54152  caffe::Net<>::ForwardFromTo()
    @     0x2b7be4b53ef1  caffe::Net<>::ForwardPrefilled()
    @     0x2b7be4b542f4  caffe::Net<>::Forward()
    @     0x2b7be4a83a83  caffe::Net<>::ForwardBackward()
    @     0x2b7be4a791aa  caffe::Solver<>::Step()
    @     0x2b7be4a78bd2  caffe::Solver<>::Solve()
    @           0x4116a1  train()
    @           0x41346c  main
    @     0x2b7be649176d  (unknown)
    @           0x410b29  (unknown)

**Comment by SarahwXU** [comment]
Hello,
Got the same problem when I run the training demo (CamVid) in the release mode.
